src/route/dataset-route.ts

```diff
@@ -39,72 +48,81 @@ function checkDatasetID(datasetID: string, res: Response): boolean {
+                    {
+                        lang: ENGLISH,
+                        message: t(translationString, { lng: ENGLISH })
+                    },
+                    {
+                        lang: WELSH,
+                        message: t(translationString, { lng: WELSH })
                     }
+                ],
+                tag: {
+                    name: translationString,
+                    params: {}
                 }
-            ]
-        };
+            }
+        ]
+    };
+}
+
+
+apiRoute.post('/', upload.single('csv'), async (req: Request, res: Response) => {
+    if (!req.file) {
         res.status(400);
-        res.json(err);
+        res.json(errorDtoGenerator('csv', 'errors.no_csv_data'));
         return;
     }
     const lang: string = req.body?.language || req.i18n.language;
     const title: string = req.body?.title;
     if (!title) {
-        const err: ViewErrDTO = {
-            success: false,
-            dataset_id: undefined,
-            errors: [
-                {
-                    field: 'title',
-                    message: [
-                        {
-                            lang: ENGLISH,
-                            message: t('errors.no_title', { lng: ENGLISH })
-                        },
-                        {
-                            lang: WELSH,
-                            message: t('errors.no_title', { lng: WELSH })
-                        }
-                    ],
-                    tag: {
-                        name: 'errors.no_title',
-                        params: {}
-                    }
-                }
-            ]
-        };
         res.status(400);
-        res.json(err);
+        res.json(errorDtoGenerator('title', 'errors.no_title'));
         return;
     }
-    const dataset = Dataset.createDataset(title, 'BetaUser');
-    const saved_dataset_record = await dataset.save();
-    saved_dataset_record.addTitleByString(title, lang);
-    const uploadDTO = await uploadCSVToBlobStorage(req.file?.buffer, saved_dataset_record);
-    if (!uploadDTO.success) {
-        res.status(400);
+    let importRecord: Import;
+    try {
+        importRecord = await uploadCSVToBlobStorage(req.file?.stream, req.file?.mimetype);
+    } catch (err) {
+        logger.error(`An error occured trying to upload the file with the following error: ${e}`);
+        res.status(500);
+        res.json({ message: 'Error uploading file' });
+        return;
     }
+
+    // Everything looks good so far, let's create the dataset and revision records
+    const dataset = new Dataset();
+    dataset.creation_date = new Date();
+    // TODO change how we handle authentication to get the user on the Backend
+    dataset.created_by = 'Test User';
+    const saved_dataset_record = await dataset.save();
+    const datasetInfo = new DatasetInfo();
+    datasetInfo.language = lang;
+    datasetInfo.title = title;
+    datasetInfo.dataset = saved_dataset_record;
```

Comment:
This needs a flag to mark it as not-yet translated and which language is now the authoritative/current version of the text value.

Reply:
No... We know if things have been translated based on the number of records attached to each dataset... In their draft form theres always one... English or Welsh.  When we look again to see if the translation step has been completed we are looking for 2 instances of datasetInfo to be present... One in English and one in Welsh:

```
datasetInfo: [
  {
    datasetId: '12345678-1234-1234-1234-123456789012',
    language: 'en-GB',
    title: 'English Title'
  }, {
    datasetId: '12345678-1234-1234-1234-123456789012',
    language: 'cy-GB',
    title: 'Welsh Title'
  }
]
```

We can get everything from the database about what has and has not been completed without using flags to tell us.

Reply:
That covers the initial state.
But in the steady state there are two translations, both related.
Then a change happens and there are still two translations, one old and one new and we don't know which is which given the metadata.